src/opers.cc

```diff
@@ -757,28 +790,9 @@ static Obj FuncWITH_HIDDEN_IMPS_FLAGS(Obj self, Obj flags)
       }
     }
 
-    /* add to hash table, shuffling old values along (last one falls off) */
-    hash = base_hash;
-
-    new_with = with;
-    new_flags = flags;
-
-    for (hash_loop = 0; hash_loop < 3; ++hash_loop) {
-        old_flags = ELM_PLIST(WITH_HIDDEN_IMPS_FLAGS_CACHE, hash * 2 + 1);
-        old_with = ELM_PLIST(WITH_HIDDEN_IMPS_FLAGS_CACHE, hash * 2 + 2);
+    StoreHashTable<HIDDEN_IMPS_CACHE_LENGTH>(WITH_HIDDEN_IMPS_FLAGS_CACHE,
+                                             hash, with, flags);
```

Comment:
you pass in 'hash, with, flags' here (and other places), but the new function takes 'hash, flags, with).

Reply:
Oh wow. It's horrifying that this even runs